systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -584,9 +584,11 @@ def hasIP(self, ip):
         return ip in self.address.values()
 
     def arpPing(self):
+        if not self.address['gateway'] or self.address['gateway'] == 'None':
+            return
         cmd = "arping -c 1 -I %s -A -U -s %s %s" % (
             self.dev, self.address['public_ip'], self.address['gateway'])
-        CsHelper.execute(cmd)
+        CsHelper.execute(cmd, wait=False)
```

Comment:
Flag arguments are an anti-pattern.  Please consider introducing and ``executeAndWait`` function instead.